backends/tfhe-cuda-backend/cuda/src/device.cu

```diff
@@ -252,3 +252,31 @@ int cuda_get_max_shared_memory(uint32_t gpu_index) {
 }
 
 void cuda_synchronize_stream(cuda_stream_t *stream) { stream->synchronize(); }
+
+/// Copy from src to dest size bytes return next 1 byte pointer
+void *cuda_concatenate_device_memory_async(void *dest, void *src, uint64_t size,
+                                           cuda_stream_t *stream) {
+  if (size == 0)
+    return dest;
+  cudaPointerAttributes attr_dest;
+  check_cuda_error(cudaPointerGetAttributes(&attr_dest, dest));
+  if (attr_dest.device != stream->gpu_index &&
+      attr_dest.type != cudaMemoryTypeDevice) {
+    PANIC("Cuda error: invalid dest device pointer in copy from GPU to GPU.");
+  }
+  cudaPointerAttributes attr_src;
+  check_cuda_error(cudaPointerGetAttributes(&attr_src, src));
+  if (attr_src.device != stream->gpu_index &&
+      attr_src.type != cudaMemoryTypeDevice) {
+    PANIC("Cuda error: invalid src device pointer in copy from GPU to GPU.");
+  }
+  if (attr_src.device != attr_dest.device) {
+    PANIC("Cuda error: different devices specified in copy from GPU to GPU.");
+  }
+
+  check_cuda_error(cudaSetDevice(stream->gpu_index));
+  check_cuda_error(cudaMemcpyAsync(dest, src, size, cudaMemcpyDeviceToDevice,
+                                   stream->stream));
```

Comment:
This code is duplicate with `cuda_memcpy_async_gpu_to_gpu`: I don't really understand the purpose of this entry point, it copies src to dest and then returns dest shifted by size? So what's the purpose of the copy? And why adding an entry point, it seems to me it would be easier to read if we just called memcpy_gpu_to_gpu followed by a pointer shift on the Rust side.